src/main/java/com/javadiscord/javabot/other/Database.java

```diff
@@ -169,253 +145,245 @@ public Document guildDoc (Guild guild) {
-
-            return var;
-
-        } catch (Exception e) {
-
-            if (guildDocExists(guild)) insertGuildDoc(guild);
-            return 0;
+            return guild.getTextChannelById(id);
+        } catch (NumberFormatException e) {
+            return null;
         }
     }
 
-    public boolean getConfigBoolean (Guild guild, String path) {
-
-        MongoDatabase database = mongoClient.getDatabase("other");
-        MongoCollection<Document> collection = database.getCollection("config");
-
+    public Role getConfigRole(Guild guild, String path) {
+        String id = getConfigString(guild, path);
         try {
-            String doc = collection.find(eq("guild_id", guild.getId())).first().toJson();
-            String[] splittedPath = path.split("\\.");
-
-            JsonObject root = JsonParser.parseString(doc).getAsJsonObject();
-            for (int i = 0; i < splittedPath.length - 1; i++) root = root.get(splittedPath[i]).getAsJsonObject();
-            boolean var = root.get(splittedPath[splittedPath.length - 1]).getAsBoolean();
-
-            return var;
-
-        } catch (Exception e) {
-
-            if (guildDocExists(guild)) insertGuildDoc(guild);
-            return false;
+            return guild.getRoleById(id);
+        } catch (NumberFormatException e) {
+            return null;
         }
     }
 
-    public TextChannel getConfigChannel (Guild guild, String path) {
-
+    public String getConfigChannelAsMention(Guild guild, String path) {
         String id = getConfigString(guild, path);
-        return guild.getTextChannelById(id);
+        try {
+            return guild.getTextChannelById(id).getAsMention();
+        } catch (NumberFormatException e) {
+            return "None";
+        }
     }
 
-    public Role getConfigRole (Guild guild, String path) {
-
+    public String getConfigRoleAsMention(Guild guild, String path) {
         String id = getConfigString(guild, path);
-        Role role;
-
         try {
-            role = guild.getRoleById(id);
-        } catch (Exception e) {
-            role = null;
+            return guild.getRoleById(id).getAsMention();
+        } catch (NumberFormatException e) {
+            return "None";
         }
+    }
+
+    private void querySB(String guildId, String channelId, String messageId, String prop, Object newValue) {
+        Document setData = new Document(prop, newValue);
+        Document update = new Document("$set", setData);
 
-        return role;
+        Document query = new Document("guild_id", guildId)
+                .append("channel_id", channelId)
+                .append("message_id", messageId);
+        mongoClient.getDatabase("other")
+                .getCollection("starboard_messages")
+                .updateOne(query, update);
     }
 
-    public String getConfigChannelAsMention (Guild guild, String path) {
+    public void querySBDString(String gID, String cID, String mID, String value, String newValue) {
```

Comment:
For this and the other `...SB` methods, it would be nice to refactor their names to be `...StarBoard` instead, so that it's actually readable for everyone.

Reply:
sure